generate_analytical_dataset.py

```diff
@@ -19,26 +19,34 @@
 def main():
     response_dict = retrieve_secrets()
     s3_publish_bucket = get_publish_bucket(response_dict)
+    collections_all = get_collections_dict(response_dict)
     published_database_name = get_published_db_name()
     database_name = get_staging_db_name()
     spark = get_spark_session()
     tables = get_tables(database_name)
+
     for table_to_process in tables:
-        adg_hive_table = database_name + "." + table_to_process
-        adg_hive_select_query = "select * from %s" % adg_hive_table
-        df = get_dataframe_from_staging(adg_hive_select_query, spark)
-        keys_map = {}
-        values = (
-            df.select("data")
-                .rdd.map(lambda x: getTuple(x.data))
-                # TODO Is there a better way without tailing ?
-                .map(lambda decrypted_db_obj: get_plaintext_key_calling_dks(decrypted_db_obj,keys_map))
-                .map(lambda decrypted_db_obj: decrypt(decrypted_db_obj))
-                .map(lambda decrypted_db_obj: validate(decrypted_db_obj))
-                .map(lambda decrypted_db_obj: sanitize(decrypted_db_obj))
-        )
-        parquet_location = persist_parquet(s3_publish_bucket, table_to_process, values)
-        create_hive_on_published(parquet_location, published_database_name, spark, table_to_process)
+        collection_name = table_to_process.replace('_hbase','')
+        if collection_name in collections_all:
+            adg_hive_table = database_name + "." + table_to_process
+            adg_hive_select_query = "select * from %s" % adg_hive_table
+            df = get_dataframe_from_staging(adg_hive_select_query, spark)
+            keys_map = {}
+            values = (
+                df.select("data")
+                    .rdd.map(lambda x: getTuple(x.data))
+                    # TODO Is there a better way without tailing ?
+                    .map(lambda decrypted_db_obj: get_plaintext_key_calling_dks(decrypted_db_obj,keys_map))
+                    .map(lambda decrypted_db_obj: decrypt(decrypted_db_obj))
+                    .map(lambda decrypted_db_obj: validate(decrypted_db_obj))
+                    .map(lambda decrypted_db_obj: sanitize(decrypted_db_obj))
+            )
+            parquet_location = persist_parquet(s3_publish_bucket, collection_name, values)
+            prefix = "${file_location}/" + collection_name, collection_name + ".parquet"
+            tag_objects(s3_publish_bucket, prefix, collection_name, collections_all)
```

Comment:
Should this tag_objects be called in the for loop ? The logic of the tag objects is iterating all the parquet files anyways so  can the tag objects be called outside for loop  ?